File: 2022/Python/Day 1/AoC-Day1-Part 1.py
def D1P1(Input):
    input_file = open(Input,"r")
    InputList = input_file.read().splitlines()

    elf_list = []

    current = 0
    for line in InputList:
      if line != '':
        current += int(line)
      else:
        elf_list.append(current)
        current = 0
    elf_list.append(current)
    elf_list.sort(reverse=True)
    return elf_list[0]

assert D1P1("D1 Test.txt") == 24000, "Function does not pass example given"

if __name__ == "__main__":
    print(D1P1("D1 Input.txt"))

File: 2022/JavaScript/Day 1/AoC-Day1-Part1.js
const {readFileSync } = require('fs');

const D1P1 = (input)=>{
  const contents = readFileSync(input, 'utf-8');

  let arr = contents.split(/\n\n/)

  arr = arr.map(str => str.split('\n'))

  arr = arr.map(elf => elf.reduce( (acc,curr) => acc + Number(curr),0)).sort( (a,b) =>  b - a)

  return arr.at(0)

}

const assert = (condition) => {
  if (!condition){
    throw Error('Function does not pass example given')
  }
}

assert(D1P1('./D1 Test.txt')===24000)

console.log(D1P1('./D1 Input.txt'))

